.kokoro/tests/run_tests.sh

```diff
@@ -21,12 +21,28 @@ shopt -s globstar
 
 DIFF_FROM=""
 
-# `--only-diff-master will only run tests on project changes from the master branch.
+# We need to temporarily turn off -e for detecting changes in the test
+# driver.
+set +e
+
+# `--only-diff-master` will only run tests on project changes on the
+# last common commit from the master branch.
 if [[ $* == *--only-diff-master* ]]; then
-    DIFF_FROM="origin/master.."
+    git diff --quiet "origin/master..." .kokoro/tests .kokoro/docker \
+	.kokoro/trampoline_v2.sh
+    CHANGED=$?
```

Comment:
nit: I think this would be easier to read (and get away without a comment) if you limit it to the immediate area
```suggestion
    set +e
    git diff --quiet "origin/master..." .kokoro/tests .kokoro/docker \
	.kokoro/trampoline_v2.sh
    CHANGED=$?
    set -e
```

Reply:
Done with the next commit